### PR TITLE
Bug 1963871: bump(*): Update the go version to 1.16

### PR DESCRIPTION
**Pull request by retroflexer**
Bumping the go version to 1.16.

### CODE REVIEW

**Review by lilic — COMMENTED** [review]
/lgtm

/retest

**Comment by lilic** [comment]
I believe we want to merge https://github.com/openshift/release/pull/18621 first, before bumping to go 1.16.

File: .ci-operator.yaml [hunk 1]
```diff
@@ -1,4 +1,4 @@
 build_root_image:
   name: release
   namespace: openshift
-  tag: rhel-8-release-golang-1.15-openshift-4.8
+  tag: rhel-8-release-golang-1.16-openshift-4.8
```
Comment (lilic) [inline .ci-operator.yaml:4]: See https://github.com/openshift/cluster-etcd-operator/pull/607

We need 
```suggestion
  tag: golang-1.16
```

**Review by lilic — COMMENTED** [review]
Based on new information from testplatform team.

File: go.mod [hunk 2]
```diff
@@ -1,6 +1,6 @@
 module github.com/openshift/cluster-etcd-operator
 
-go 1.15
+go 1.16
```
Comment (sttts) [inline go.mod:3]: this needs a `go mod tidy` call. go.sum is expected to change.

File: .ci-operator.yaml (hunk 1 above)
Comment (sttts) [reply .ci-operator.yaml:4]: actually this one might work. ckao uses the same.

File: go.mod (hunk 2 above)
Comment (lilic) [reply go.mod:3]: Surprised tests passed, maybe we should add in next release a go mod test as well.

File: .ci-operator.yaml (hunk 1 above)
Comment (retroflexer) [reply .ci-operator.yaml:4]: I left it as is to match with ckao and pushed the change with go.sum.

File: go.mod (hunk 2 above)
Comment (retroflexer) [reply go.mod:3]: Pushed the new go.sum.

**Pull request merged**